test/e2e/common.go

```diff
@@ -243,16 +243,17 @@ func EnsureControlPlaneInitialized(ctx context.Context, input clusterctl.ApplyCl
 	Eventually(func() error {
 		return getter.Get(ctx, key, kubeadmControlPlane)
 	}, input.WaitForControlPlaneIntervals...).Should(Succeed(), "Failed to get KubeadmControlPlane object %s/%s", cluster.Spec.ControlPlaneRef.Namespace, cluster.Spec.ControlPlaneRef.Name)
+	controlPlane := discoveryAndWaitForControlPlaneInitialized(ctx, input, result)
```

Comment:
This probably can't run before the Helm charts are installed because it checks for the Machines to be "Ready" and that won't happen until the nodes have Provider IDs attached to them so it needs cloud-provider to be installed first

Reply:
/lgtm cancel

Reply:
Indeed.

Though interestingly the flakes I was observing were during the azuredisk-csi helm chart installation, which we do after cloud-provider-azure. So... we'll install *that* helm chart after validating control plane readiness.